xarray/core/dataset.py

```diff
@@ -5617,6 +5808,39 @@ def dropna(
             Which variables to check for missing values. By default, all
             variables in the dataset are checked.
 
+        Examples
+        --------
+
+        >>> dataset = xr.Dataset(
+        ...     {
+        ...         "temperature": (
+        ...             ["time", "location"],
+        ...             [[23.4, 24.1], [np.nan, 22.1], [21.8, 24.2]],
+        ...         )
+        ...     },
+        ...     coords={"time": [1, 2, 3], "location": ["A", "B"]},
+        ... )
+        >>> dataset
+        <xarray.Dataset>
+        Dimensions:      (time: 3, location: 2)
+        Coordinates:
+          * time         (time) int64 1 2 3
+          * location     (location) <U1 'A' 'B'
+        Data variables:
+            temperature  (time, location) float64 23.4 24.1 nan 22.1 21.8 24.2
+
+        # Drop NaN values from the dataset
```

Comment:
Good example, but we can do more here. There are 4 different keyword arguments to this method that we could demonstrate the difference between.

Let's use a dataset with multiple variables to show off the fact we can do this on a `Dataset` not just `DataArray`.